### PR TITLE
feat(autocomplete): add isOpen attribute

### PR DESCRIPTION
**Pull request by josephperrott**
For #5881

### CODE REVIEW

File: src/lib/autocomplete/autocomplete.ts [hunk 1]
```diff
@@ -57,6 +57,11 @@ export class MdAutocomplete implements AfterContentInit {
   /** Whether the autocomplete panel should be visible, depending on option length. */
   showPanel = false;
 
+  _isOpen: boolean = false;
```
Comment (crisbeto) [inline src/lib/autocomplete/autocomplete.ts]: I'm not sure we need this extra flag. `isOpen` should the same as `this.showPanel && this.options.length > 0`.

Comment (josephperrott) [reply src/lib/autocomplete/autocomplete.ts]: That was my initial thought as well, however, `showPanel` is not set to false on `blur`, which prevents it from working that way.

File: src/demo-app/autocomplete/autocomplete-demo.html [hunk 2]
```diff
@@ -66,19 +66,6 @@
   </md-card>
 </div>
 
-<md-autocomplete #reactiveAuto="mdAutocomplete" [displayWith]="displayFn">
```
Comment (crisbeto) [inline src/demo-app/autocomplete/autocomplete-demo.html]: Why were these removed from the demo?

File: src/lib/autocomplete/autocomplete.ts (hunk 1 above)
Comment (crisbeto) [reply src/lib/autocomplete/autocomplete.ts]: Makes sense.

File: src/demo-app/autocomplete/autocomplete-demo.html (hunk 2 above)
Comment (josephperrott) [reply src/demo-app/autocomplete/autocomplete-demo.html]: They are duplicates of the same elements above.

It look slike we had 5 `<md-autocomplete>`s for 3 `[mdAutocompleteTrigger]`s

File: src/lib/autocomplete/autocomplete.ts (hunk 1 above)
Comment (crisbeto) [reply src/lib/autocomplete/autocomplete.ts]: Cool, also a final nit: can you add a doc string to the `isOpen` getter so it shows up in the docs?

Comment (josephperrott) [reply src/lib/autocomplete/autocomplete.ts]: Yeah, not sure how I deleted the doc line I had for it.
